### ISSUE TITLE
Tethering + Orbot

### ISSUE THREAD

**Issue by vpr2k1** [ukanth/afwall#940]
Hello Ukanth

I try to use wifi Tethering and afw + Orbot at the same time, notebook connects to android via wifi, afw white list checkboxes settled a right way.

Override DNS ( for DNS change, actually i can use tethering with afw only when i have reconfigured DNS to Google or Open DNS and write them on Network Adapter DNS settings)
At android all works great, all trafic pass trought Orbot, when Orbot is down - packets is rejected/droped, all works great

But when i try to use tethering with Orbot, it doesnt work, best of i already achieved - it's a possibilty use internet on PC via wifi tethering with working AFW WITHOUT passing traffic trought orbot.
I don't know why it's happen.
When i try to set Orbot checkbox on tethering - now, on 3.1.0 version, it set correct, without "Error applying iptables rules" like before. But still no result

Orbot option - "Open proxy on all interfaces" is ON

I suppose it's a great, useful feature to use proxyfication via Orbot and tethering, and many people will appreciate it

![screenshot_2019-01-28-13-16-25-692_org torproject android](https://user-images.githubusercontent.com/47103280/51830758-a4e63300-2301-11e9-9a02-214a30fb6475.png)
![screenshot_2019-01-28-12-26-38-438_dev ukanth ufirewall](https://user-images.githubusercontent.com/47103280/51830759-a57ec980-2301-11e9-87c5-e3d0710a7449.png)
![screenshot_2019-01-28-12-22-41-812_dev ukanth ufirewall](https://user-images.githubusercontent.com/47103280/51830760-a57ec980-2301-11e9-9f2e-fa8bbf6d41bb.png)
![screenshot_2019-01-28-12-22-28-012_dev ukanth ufirewall](https://user-images.githubusercontent.com/47103280/51830762-a6176000-2301-11e9-816a-3575f9e97e4a.png)

**Comment by Jookia** [comment]
Does Orbot use iptables for the tethering? If so it would conflict with AFWall.

**Comment by shiftyshady** [comment]
AFAIK, no, not any more.
Orbot can work as a traditional Android vpn, or it can also be a socks proxy that other apps can be set up to connect to (e.g.: Orfox)

I wonder... what the heck does the Tor checkbox do? Is it documented somewhere?

OP, I think the way to torify local phone apps through Tor while having a hotspot is to leave Tor in NON-VPN move and use a AFWall script to redirect traffic of the apps you want torified to tor.
I am stil  trying to find a script that works

**Comment by Jookia** [comment]
The Tor checkbox sends all of the application's allowed traffic through Tor

**Comment by vpr2k1** [comment]
> OP, I think the way to torify local phone apps through Tor while having a hotspot is to leave Tor in NON-VPN move and use a AFWall script to redirect traffic of the apps you want torified to tor.
> I am stil trying to find a script that works

I was trying to find some, but no result, and i have no sufficient skills for do it myself, but like i say in first post I suppose it's a great, useful feature to use proxyfication via Orbot and tethering together, something like relocatable proxy hot-spot) i think it will be nice

**Comment by shiftyshady** [comment]
> 
> 
> The Tor checkbox sends all of the application's allowed traffic through Tor

Yes!, I tested the tor checkbox on another phone and I can see that it works, even when hotspot is enabled.

Edit:
On my main phone the Tor checkbox only works if I allow "(Any app) - Same as selecting all apps", does this mean it is a DNS problem?.

@vpr2k1 does the Tor checkbox works for you if you allow "(Any app) - Same as selecting all apps"?

**Comment by vpr2k1** [comment]
@shiftyshady
Yes, tor works fine with Any app checkbox, and actually i don't have any problem with magisk, startup scripts and other feature..
I can't get how it's regard to issue?

**Comment by Malsasa** [comment]
I also find same issue as I cannot get connected to Tor Network on laptop (check.torproject.org says "Sorry") while connected both via Wifi Hotspot and USB Tethering to my phone. I have checked "Open proxies on all interfaces" so I expect my tethered laptop to automatically (transparently) connected to internet over Tor. On my Android phone, I use Orbot 16.1.2 from F-Droid. Is there anything lacking in my report? I wish soon I can run "Tor Tethering" for my laptop. Thanks.
